### DISCUSSION TITLE
Prisma 1: filtering with OR

### DISCUSSION THREAD

**Discussion by dimaip** [prisma/prisma#3030] [Q&A] [answered]
I'm using Prisma 1 with nexus and MongoDB.

I have this resolver for getting a list of things:

```js
t.list.field('things', {
    type: 'Thing',
    nullable: false,
    args: {},
    resolve: async (_, args, ctx) => {
      return ctx.prisma.things({
        where: {
          OR: [],
        },
      });
    },
  });
```

I get the following error when I try to access it:

`Variable '$where' expected value of type 'ThingWhereInput' but got: {\"OR\":[]}. Reason: 'OR' Field 'OR' is not defined in the input type 'ThingWhereInput'`.

If I lookup the generated nexus.ts, I indeed do not see `OR` on it (however there's `AND`).

I have noticed that other types that end with `ScalarWhereInput` do have `OR` and `NOT`.

I'm trying to understand for which Prisma types the scalar definitions are generated, and it seems for those that  don't have any `@relation`s in them.

So does it mean that I cannot have `OR` queries on types that have relations? Why? I don't think the docs say anything about it.

**Comment by dimaip** [comment]
Update: it doesn't have anything to do with `@relation`.

I tried to create a minimal example:

```
type Thing @db(name: "myMongoCollection") {
  _id: ID! @id
}
```

And it generates this:
```
export interface ThingWhereInput {
  _id?: string | null
  _id_not?: string | null
  _id_in?: string[]
  _id_not_in?: string[]
  _id_lt?: string | null
  _id_lte?: string | null
  _id_gt?: string | null
  _id_gte?: string | null
  _id_contains?: string | null
  _id_not_contains?: string | null
  _id_starts_with?: string | null
  _id_not_starts_with?: string | null
  _id_ends_with?: string | null
  _id_not_ends_with?: string | null
  AND?: ThingWhereInput[]
}
```

While some other types are generated as this:
```
export interface RoleScalarWhereInput {
  _id?: string | null
  _id_not?: string | null
  _id_in?: string[]
  _id_not_in?: string[]
  _id_lt?: string | null
  _id_lte?: string | null
  _id_gt?: string | null
  _id_gte?: string | null
  _id_contains?: string | null
  _id_not_contains?: string | null
  _id_starts_with?: string | null
  _id_not_starts_with?: string | null
  _id_ends_with?: string | null
  _id_not_ends_with?: string | null
  name?: string | null
  name_not?: string | null
  name_in?: string[]
  name_not_in?: string[]
  name_lt?: string | null
  name_lte?: string | null
  name_gt?: string | null
  name_gte?: string | null
  name_contains?: string | null
  name_not_contains?: string | null
  name_starts_with?: string | null
  name_not_starts_with?: string | null
  name_ends_with?: string | null
  name_not_ends_with?: string | null
  AND?: RoleScalarWhereInput[]
  OR?: RoleScalarWhereInput[]
  NOT?: RoleScalarWhereInput[]
}
```

**Comment by dimaip** [comment, marked as answer]
OR and NOT support have been silently removed from the Mongo version of Prisma, without even touching the docs! Great job! :(
https://github.com/prisma/prisma1/issues/3897

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
